### ISSUE TITLE
Unable to run GParted

### ISSUE THREAD

**Issue by bala7s** [pop-os/cosmic-epoch#1392]
As you can see in the screenshot I can't launch GParted.

![screenshot-2024-12-22-10-07-50](https://github.com/user-attachments/assets/450e158e-2bbc-4f4f-b10e-522f9990d82d)

**Comment by git-f0x** [comment]
Apps can't run as root on Wayland, so Gparted doesn't work.
You can reduce security and make Gparted work by using the following command: `xhost +Sl:localuser:root`
But it would be better to use another disk management tool (e.g. GNOME Disks).

**Comment by jokeyrhyme** [comment]
Yeah, applications that split their functionality into a non-root part that includes the UI and a separate root process that performs privileged tasks is the recommendation, even for X11 apps

GUI toolkits are often giant blobs of C/Cxx and you want as little of that as possible to be running with root powers

**Comment by HackintoshwithUbuntu** [comment]
That's interesting, I guess it's slightly more confusing for newer users since Gparted runs fine off of the Pop!_Os live usb.

**Comment by jokeyrhyme** [comment]
Oh, I just remembered, I think there's a -E or --preserve-environment flag for sudo that might help
Otherwise, the root process doesn't have access to the DISPLAY environment variable

**Comment by HackintoshwithUbuntu** [comment]
I can confirm `sudo -E gparted` and `sudo --preserve-env gparted` work to start gparted on Pop_os with Cosmic (running Wayland).

**Comment by LiamillionSS** [comment]
Duplicate to https://github.com/pop-os/cosmic-comp/issues/731

**Comment by gene1wood** [comment]
`gparted` launched from the Pop!_OS Launcher silently fails, but running `sudo --preserve-env gparted` in a terminal works.

**Comment by usbeehu** [comment]
Yup, that works. As a workaround I picked KDE Partition Manager that works properly under Wayland out of the box.
